services/alertmanager/alertmanager.go

```diff
@@ -35,12 +35,17 @@ import (
 )
 
 const (
-	resendInterval      = 30 * time.Second
-	alertmanagerDataDir = "/srv/alertmanager/data"
-	prometheusDir       = "/srv/prometheus"
-	path                = "/srv/alertmanager/alertmanager.base.yml"
+	// Environment variable to overwrite resendInterval during testing
+	envInterval = "PERCONA_TEST_ALERTMANAGER_INTERVAL"
+
+	resolveTimeoutFactor = 3
+	alertmanagerDataDir  = "/srv/alertmanager/data"
+	prometheusDir        = "/srv/prometheus"
+	path                 = "/srv/alertmanager/alertmanager.base.yml"
 )
 
+var resendInterval = 30 * time.Second
```

Comment:
Let's make it a `Service` field, not a package-level variable.

Reply:
I had thought of that first but since the EndsAt is added in a `Registry` method, I couldn't think of how will `CreateAlert` acquire the value of the `resendInterval` if it were a field of `Service` cc @AlekSi

Reply:
Then it should be in a registry.go and set during process initialisation.
The problem with the current code that `New` "constructor" also changes global state. That's unexpected.